src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public Figure getRandomFigure() {
+        GeometricFigure figure = new GeometricFigureSupplier().getRandomFigure();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random random = new Random();
```

Comment:
The 'Random' instance should be a class-level variable to avoid creating a new instance each time the method is called.